hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/master/BackupLogCleaner.java

```diff
@@ -176,12 +187,29 @@ protected static boolean canDeleteFile(Map<Address, Long> addressToLastBackupMap
       Address walServerAddress = Address.fromString(hostname);
       long walTimestamp = AbstractFSWALProvider.getTimestamp(path.getName());
 
-      if (
-        !addressToLastBackupMap.containsKey(walServerAddress)
-          || addressToLastBackupMap.get(walServerAddress) >= walTimestamp
-      ) {
+      if (!addressToLastBackupMap.containsKey(walServerAddress)) {
+        if (LOG.isDebugEnabled()) {
+          LOG.debug("No backup found for server: {}. Deleting file: {}",
+            walServerAddress.getHostName(), path);
+        }
         return true;
       }
+
+      Long lastBackupTs = addressToLastBackupMap.get(walServerAddress);
+      if (lastBackupTs > walTimestamp) {
+        if (LOG.isDebugEnabled()) {
+          LOG.debug(
+            "Backup found for server: {}. Backup from {} is newer than file, so deleting: {}",
```

Comment:
You could make operators' lives easier by including human readable timestamps in these log lines that are intended for humans. I think the whole backup system would benefit from this.

Reply:
You don't have your phone configured in epoch millis??

Joking aside, I see your point here, but I actually often get annoyed when _only_ more readable times are logged because then I need to think about timezones, I inevitably need to convert to millis if the investigation progresses, ... I could be convinced to log both, but that feels superfluous? Maybe I'm too comfortable with my "epoch to readable datetime" alfred workflow

Reply:
In my experience, operators are very accustomed to thinking in UTC, no matter where they live. What I'm saying is that I expect an operator/SRE to be able to read UTC timestamps as tablestakes. A human being reading milliseconds since unix epoc, less so.

But anyway, just an observation, not a blocking suggestion; this code (as in, all of hbase) is littered with logging epoc millis.